### PR TITLE
Removing PyQuante2 for packaging

### PR DESCRIPTION
**Pull request by FNTwin**
- Removed pyquante2 for packaging in pip 
- Reworked pyproject.toml and added infos (added importlib in dependencies to avoid issues)
- Added dev dynamic versioning
- Added a pip install for pyquante2 in the docs and the from_pyquante function to enable notebook use

### CODE REVIEW

**Review by hatemhelal — CHANGES_REQUESTED** [review]
No issues with removing the dependency for packaging but I'd prefer a more incremental change to not lose functionality and testing for dev.  

Separately we should add support for looking up either experimentally or DFT minimised geometries from a known good database (PubChem might allow this?). this would let us part with the `from_pyquante` function that is used in the notebook and benchmark.

File: test/test_benchmark.py [hunk 1]
```diff
@@ -9,24 +9,10 @@
     nuclear_basis,
     overlap_basis,
 )
-from mess.interop import from_pyquante
 from mess.structure import molecule
-from mess.zeropad_integrals import overlap_basis_zeropad
 from conftest import is_mem_limited
```
[the remark this answers is not in the document]
Comment (FNTwin) [reply test/test_benchmark.py:15]: As I'll add the lazy import of pyquante I agree with this

File: mess/interop.py [hunk 2]
```diff
@@ -35,19 +34,3 @@ def from_pyscf(mol: "gto.Mole") -> Tuple[Structure, Basis]:
     basis = basisset(structure, basis_name=mol.basis)
 
     return structure, basis
-
-
-def from_pyquante(name: str) -> Structure:
```
[the remark this answers is not in the document]
Comment (FNTwin) [reply mess/interop.py:40]: Agreed. Pyquante2 will still be out of the environment (the pypi one)

**Comment by FNTwin** [comment]
> Separately we should add support for looking up either experimentally or DFT minimised geometries from a known good database (PubChem might allow this?). this would let us part with the from_pyquante function that is used in the notebook and benchmark.

Agreed but this should be tackled into another issue. This is trying to tackle the packaging issues.

**Review by hatemhelal — APPROVED** [review]
:shipit:

**Pull request merged**